enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,974 @@
+manually specify `IngressController` role subnets.
+
+The existing validation for the deprecated `Subnets` field should apply to all subnets (`IngressController`
+and `Cluster`) specified in `SubnetsConfig`, such as the [existing validation](https://github.com/openshift/installer/blob/0d77aa8df5ddc68e926aa11da24a87981021b256/pkg/asset/installconfig/aws/subnet.go#L91)
+that confirms all subnets are from the same VPC.
+
+Since the AWS Cloud Controller Manager (CCM) will reject certain subnets, the installer will need to add
+additional validation for `IngressController` role subnets:
+
+- The installer should reject multiple `IngressController` role subnets in the same AZ as this will be rejected
+  by the AWS CCM.
+
+#### IngressConfig API Updates
+
+The `spec.loadBalancer.platform.aws.classicLoadBalancer.subnets` and `spec.loadBalancer.platform.aws.networkLoadBalancer.subnets`
+fields will be added to the IngressConfig to encode the default subnets for all IngressControllers. Regardless of the
+load balancer type specified in the `platform.aws.lbType` install-config field, the installer will populate both
+`networkLoadBalancer` and `classicLoadBalancer` subnet fields with the values provided in the new
+`platform.aws.subnetsConfig` install-config field. The `AWSSubnets` struct is duplicated from the IngressController CRD
+to maintain API consistency. Although the new `subnetsConfig` install-config field only supports specifying subnets by
+ID, the IngressConfig CRD will still support specifying subnets by name for completeness.
+
+```go
+// AWSIngressSpec holds the desired state of the Ingress for Amazon Web Services infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+// +union
+type AWSIngressSpec struct {
+    [...]
+    // classicLoadBalancerParameters holds configuration parameters for an AWS
+    // classic load balancer.
+    //
+    // +optional
+    ClassicLoadBalancerParameters *AWSClassicLoadBalancerParameters `json:"classicLoadBalancer,omitempty"`
+
+    // networkLoadBalancerParameters holds configuration parameters for an AWS
+    // network load balancer.
+    //
+    // +optional
+    NetworkLoadBalancerParameters *AWSNetworkLoadBalancerParameters `json:"networkLoadBalancer,omitempty"`
+}
+
+// AWSClassicLoadBalancerParameters holds configuration parameters for an
+// AWS Classic load balancer.
+type AWSClassicLoadBalancerParameters struct {
+    // Subnets specifies the default subnets for all IngressControllers,
+    // including the default IngressController. This default will be overridden
+    // if subnets are specified on the IngressController.
+    // If omitted, IngressController subnets will be automatically discovered
+    // based on the kubernetes.io/cluster/<cluster-id> tag, whether it's public
+    // or private, and the availability zone.
+    //
+    // +optional
+    // +openshift:enable:FeatureGate=IngressConfigLBSubnetsAWS
```

Comment:
If classicLoadBalancer doesn't already exist, then this should be moved up 1 level

Reply:
I was talking specifically about the feature gate level. The feature gating should be only at the top most new field, it's not needed at lower levels, and, with your current design, you'd introduce the `classicLoadBalancer` as a stable field with nothing in it, as the fields inside are all tech preview, the `classicLoadBalancer` should be gated, not the fields within

Reply:
Whoops sorry, I thought the comment was the on the `classicLoadBalancer` field, not the feature gate.

Ack. That makes sense. Moving up one level. However, I do think we need to consider @miheer's simultaneous effort in which he is adding to `networkLoadBalancer` [here](https://github.com/openshift/api/pull/2043/files#diff-0beb33df12dd4ecac8d28c3fe688acc3a178de15c301325e33f586207b2c748cR165) (I think it should also have a feature gate). I suppose with https://github.com/openshift/kubernetes-sigs-controller-tools/pull/21 we can do `AND` in the feature gate markers if the scenario is that we both go TP at the same time, so I think it's an non-issue.

Reply:
Yep you're right, Miheer's gate is also at the wrong level and should be gated higher